.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,24 @@
+### Description
+
+_Please explain the changes you made here._
+_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
+_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
+
+----
+
+### Checklist (never delete this)
+
+Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below. 
+
+#### Content
+- [ ] Added or modified tests to cover changes and any new functionality
+- [ ] All tests passing on Travis 
+
+#### Form
+- [ ] Rebased, squashed and reworded to produce a single commit (exemptions may be made -- sparingly) that is _uncluttered by excess lines_ (lines that just say "addressed review", "fixed" etc left over from squashing)
```

Comment:
please note that while PR is in progress, we request that all the commits be left unsquashed...actually, I would move this part to the end of the review section and say clearly that after the review is complete, if new commits have been added,  they should now (and only now) be squashed by rebasing.